tests/unit/commands/buildcmd/test_build_context.py

```diff
@@ -1149,3 +1150,35 @@ def test_check_esbuild_warning(self, functions, should_print, mocked_click):
             mocked_click.assert_called_with(ExperimentalFlag.Esbuild, BuildContext._ESBUILD_WARNING_MESSAGE)
         else:
             mocked_click.assert_not_called()
+
+
+class TestBuildContext_exclude_warning(TestCase):
+    @parameterized.expand(
+        [
+            ([], [], False),
+            ("Function", ("Function",), True),
+            ("Function", ("NotTheSameFunction",), False),
+        ]
+    )
+    @patch("samcli.commands.build.build_context.LOG")
+    def test_check_exclude_warning(self, resource_id, exclude, should_print, log_mock):
+        build_context = BuildContext(
+            resource_identifier=resource_id,
+            template_file="template_file",
+            base_dir="base_dir",
+            build_dir="build_dir",
+            cache_dir="cache_dir",
+            cached=False,
+            clean=False,
+            parallel=False,
+            mode="mode",
+            excluded_resources=exclude,
+        )
+        with patch.object(build_context, "get_resources_to_build") as mocked_resources_to_build:
```

Comment:
is this patching required?

Reply:
Upon further testing, no, it doesn't seem like this patching is required, since `_check_exclude_warning()` can be directly called upon creation of the `BuildContext`. I will fix this in the next commit.